ktor-server/ktor-server-plugins/ktor-server-sessions/jvm/src/io/ktor/server/sessions/SessionStorage.kt

```diff
@@ -4,19 +4,14 @@
 
 package io.ktor.server.sessions
 
-import io.ktor.utils.io.*
-
 /**
  * Represents a way to [write], [read] and [invalidate] session bits.
  */
 public interface SessionStorage {
     /**
-     * Writes a session [id] using a specific [provider].
-     *
-     * This method calls the [provider] with a [ByteWriteChannel] and it is in charge of the channel's lifecycle.
-     * [provider] is in charge of writing session bits to the specified [ByteWriteChannel].
+     * Writes a session [value] for [id].
      */
-    public suspend fun write(id: String, provider: suspend (ByteWriteChannel) -> Unit)
+    public suspend fun write(id: String, value: String)
```

Comment:
I like this change because it reads better, but it's going to be a breaking change for users...

Reply:
tbh current interface and reference implementations in Ktor sources and docs https://ktor.io/docs/storages.html#custom_storage (SimplifiedSessionStorage) are overcomplicated for such a simple concept, I found myself just completely lost multiple times while reading sources and trying to implement mine

Reply:
It might be better to replace String with ByteArray for value

Reply:
@artem-zinnatullin I agree that it's overcomplicated and that's why we are takeing opportunity of major release to simplify it.

@ikorennoy Why do you prefer a byte array? Cookie is a string, all users of this API read and write strings. Are there cases when we need to use byte array?

Reply:
@rsinukov, I got this idea because I was thinking about a use case where the session is a complex object that is serialized and stored in a database, using a string probably reduces the number of serialization algorithms that can be used.

Also, this suggestion is based on this example [Custom storage](https://ktor.io/docs/storages.html#custom_storage)

Reply:
Currently, our API for serializer supports only serialization to and from a string. We may change it in the future if there would be requests for this.